oeps/oep-0016.rst

```diff
@@ -0,0 +1,261 @@
+| Title         | Bootstrap Adoption                        |
++---------------+-------------------------------------------+
+| Last Modified | 2017-05-17                                |
++---------------+-------------------------------------------+
+| Author        | Andy Armstrong <andya@edx.org>            |
++---------------+-------------------------------------------+
+| Arbiter       | Dennis Jen <djen@edx.org>                 |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Architecture                              |
++---------------+-------------------------------------------+
+| Created       | 2017-04-18                                |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+In order to accelerate development, and provide a more consistent user
+experience for our users, Open edX will adopt Bootstrap 4 to style its web
+applications. The community will be able to extend the platform using any of
+Bootstrap's collection of components, as well as use Bootstrap theming to brand
+their instances. An Open edX-supported subset of components will be provided
+that are verified to support `WCAG 2.0`_ to ensure that all new user interfaces
+conform to the `edX accessibility guidelines`_.
+
+Technology Selection
+====================
+
+The following rubric was used to assess potential technologies:
+
+* Learning curve / prior experience
+* Options for community
+* Frameworks that are opinionated vs. flexible
+* Ability to easily build accessible front-ends
+* Ease of design-framework integration
+* DIY framework vs. all-inclusive platform
+* Willingness to dedicate resources to development and maintenance
+* Ease of prototyping
+* Community project support
+* Compliance and security considerations
+* Availability of developers (hiring ability)
+* Ability to achieve business strategy
+
+In addition, there were a number of key goals that such a framework must
+address:
+
+* accessibility
+* internationalization and right-to-left languages
+* theming
+* performance
+* responsive designs
+* extensibility / ease of overriding
+* modularity / scoping
+* Open edX community interest
+* size and activity of the framework's community
+* integration with build systems
+* integration into existing edX pages
+
+Bootstrap is the industry standard framework for desigining web applications. It
+comes with a comprehensive collection of web components that are designed to be
+responsive, meaning that Bootstrap applications look good at phone, tablet,
+and desktop sizes.
+
+Bootstrap 4 is the first version to support key requirements for adoption
+within the edX platform:
+
+* Bootstrap 4 has switched to using Sass (previous versions used Less)
+* It provides a more flexible grid system that supports all of edX's layouts
```

Comment:
I was thinking of the combination of flexbox plus the improvements they describe here:

https://blog.getbootstrap.com/2015/08/19/bootstrap-4-alpha/

> **Improved grid system.** We’ve added a new grid tier to better target mobile devices and completely overhauled our semantic mixins.

> **Opt-in flexbox support is here.** The future is now—switch a boolean variable and recompile your CSS to take advantage of a flexbox-based grid system and components.